test/functional/lua/vim_spec.lua

```diff
@@ -505,6 +505,19 @@ describe('lua stdlib', function()
     ]]))
   end)
 
+  it('vim.tbl_isarray', function()
+    eq(true, exec_lua("return vim.tbl_isarray({})"))
+    eq(false, exec_lua("return vim.tbl_isarray(vim.empty_dict())"))
+    eq(true, exec_lua("return vim.tbl_isarray({'a', 'b', 'c'})"))
+    eq(false, exec_lua("return vim.tbl_isarray({'a', '32', a='hello', b='baz'})"))
+    eq(false, exec_lua("return vim.tbl_isarray({1, a='hello', b='baz'})"))
+    eq(false, exec_lua("return vim.tbl_isarray({a='hello', b='baz', 1})"))
+    eq(false, exec_lua("return vim.tbl_isarray({1, 2, nil, a='hello'})"))
+    eq(true, exec_lua("return vim.tbl_isarray({1, 2, nil, 4})"))
+    eq(true, exec_lua("return vim.tbl_isarray({nil, 2, 3, 4})"))
+    eq(true, exec_lua("return vim.tbl_isarray({1, [1.5]=2, [3]=3})"))
```

Comment:
I don't think that's a valid array (not indexed by integers), so this should return `false`.

What does openResty/luaJIT2 `table.isarray()` return for that?

Reply:
I have checked the definition. As you say, it is indeed only integers.

Reply:
The tests are good, though!

Reply:
I would also fix tbl_isarray, since it only looks at whether it is a number or not.
There are several ways to determine if it is an integer, is there a standard method used by neovim?

Reply:
`math.floor(key) == key`?

Reply:
Ok. I'll use it.

Reply:
fix done.